### PR TITLE
feat: support to create view using dsl

### PR DESCRIPTION
**Pull request by uziasferreirazup**
## Purpose
Using simple library to generate dsl

## Example dsl created:
```
screen {
            navigationBar {
                title = "Beagle StackView"
                showBackButton = true
                navigationBarItems {
                    +navigationBarItem {
                        text = ""
                        image = "informationImage"
                        action = showNativeDialog {
                            title = "StackView"
                            message = "Implements widgets on top of each other."
                            buttonText = "OK"
                        }
                    }
                }
            }
            child = stack {
                children {
                    +text {
                        text = "Text 1"
                    }.flex {
                        margin {
                             top = 5.unitReal()
                        }
                    }

                    +text {
                        text = "Text 2"
                    }
                    +text {
                        text = "Text 3"
                    }
                }
            }
        }
```

### CODE REVIEW

File: android/android-declarative/src/main/java/br/com/zup/beagle/widget/form/InputWidget.kt [hunk 1]
```diff
@@ -43,14 +46,22 @@ abstract class InputWidget : WidgetView(), StateChangeable {
         return super.setId(id) as InputWidget
     }
 
-    override fun applyAppearance(appearance: Appearance): InputWidget {
-        return super.applyAppearance(appearance) as InputWidget
+    override fun appearance(block: AppearanceAutoDslBuilder.() -> Unit): InputWidget {
```
Comment (tuliopereirazup) [inline android/android-declarative/src/main/java/br/com/zup/beagle/widget/form/InputWidget.kt:49]: Does we will not support the old way anymore?

File: android/sample/src/main/java/br/com/zup/beagle/sample/fragment/StackViewFragment.kt [hunk 2]
```diff
@@ -21,27 +21,48 @@ import android.view.LayoutInflater
...
-            children = listOf(
-                Text("Text 1").applyFlex(Flex(margin = EdgeValue(top = UnitValue(5.0, UnitType.REAL)))),
-                Text("Text 2"),
-                Text("Text 3")
-            )
-        )
+        val declarative = screen {
+            withChild(br.com.zup.beagle.widget.layout.container {
+                children {
+                    +text {
```
Comment (tuliopereirazup) [inline android/sample/src/main/java/br/com/zup/beagle/sample/fragment/StackViewFragment.kt:47]: Is this `+` required when the structure is a list?

Comment (uziasferreirazup) [reply android/sample/src/main/java/br/com/zup/beagle/sample/fragment/StackViewFragment.kt:47]: yes the library implement this by default

File: android/android-declarative/src/main/java/br/com/zup/beagle/widget/form/InputWidget.kt (hunk 1 above)
Comment (uziasferreirazup) [reply android/android-declarative/src/main/java/br/com/zup/beagle/widget/form/InputWidget.kt:49]: I can add the two ways.

**Pull request closed without merging**